### PR TITLE
Added mob and vehicle spawn protocol /Bergerkiller

### PR DESCRIPTION
**Pull request by bergerkiller**
Adds the base protocol classes for all mob and vehicle types. The mob types with additional meta data during creation (parameters) reside in separate classes for proper parsing, the others reside in 'GeneralLivingEntityProtocol'

### CODE REVIEW

File: src/main/java/org/spout/vanilla/entity/protocols/living/GeneralLivingEntityProtocol.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		} else if (c instanceof Wolf) {
+			type = 95;
+		} else if (c instanceof Mooshroom) {
+			type = 96;
+		} else if (c instanceof SnowGolem) {
+			type = 97;
+		} else if (c instanceof Ocelot) {
+			type = 98;
+		} else if (c instanceof Villager) {
+			type = 120;
```
Comment (zml2008) [inline src/main/java/org/spout/vanilla/entity/protocols/living/GeneralLivingEntityProtocol.java:89]: This is horrible

**Comment by bergerkiller** [comment]
@zml2008 Can't see why, as I discussed doing that on IRC. Or do you want separate protocol class files for every single mob?

File: src/main/java/org/spout/vanilla/entity/protocols/living/GeneralLivingEntityProtocol.java (hunk 1 above)
Comment (RoyAwesome) [inline src/main/java/org/spout/vanilla/entity/protocols/living/GeneralLivingEntityProtocol.java:89]: For reference, these numbers are stored in the entities datatable

**Comment by windy1** [comment]
Could you squash and update to current HEAD please? :)

**Comment by ZNickq** [comment]
Closed on request!

**Pull request closed without merging**